plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/nessie/TestIcebergNessieCatalogWithAuth.java

```diff
@@ -0,0 +1,118 @@
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryFailedException;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.TestTable;
+import org.testcontainers.containers.Network;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergNessieCatalogWithAuth
+        extends AbstractTestQueryFramework
+{
+    private Path tempDir;
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+            throws IOException
+    {
+        deleteRecursively(tempDir, ALLOW_INSECURE);
+    }
+
+    private String nessieUri;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Network network = Network.newNetwork();
+
+        KeycloakContainer keycloakContainer =
+                closeAfterClass(KeycloakContainer.builder().withNetwork(network).build());
+        keycloakContainer.start();
+
+        ImmutableMap<String, String> envVars = ImmutableMap.<String, String>builder()
+                .putAll(NessieContainer.defaultEnvVars())
+                .put("QUARKUS_OIDC_AUTH_SERVER_URL", KeycloakContainer.getUrlWithHostName() + "/realms/master")
+                .put("QUARKUS_OIDC_CLIENT_ID", "projectnessie")
+                .put("NESSIE_SERVER_AUTHENTICATION_ENABLED", "true")
+                .buildOrThrow();
+
+        NessieContainer nessieContainer =
+                closeAfterClass(NessieContainer.builder().withEnvVars(envVars).withNetwork(network).build());
+        nessieContainer.start();
+        nessieUri = nessieContainer.getRestApiUri();
+
+        tempDir = Files.createTempDirectory("test_trino_nessie_catalog");
+
+        return getDistributedQueryRunner(nessieUri, keycloakContainer.getAccessToken());
+    }
+
+    private DistributedQueryRunner getDistributedQueryRunner(String nessieUri, String accessToken)
+            throws Exception
+    {
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(tempDir))
+                .setIcebergProperties(
+                        ImmutableMap.of(
+                                "iceberg.file-format", "PARQUET",
+                                "iceberg.catalog.type", "nessie",
+                                "iceberg.nessie-catalog.uri", nessieUri,
+                                "iceberg.nessie-catalog.default-warehouse-dir", tempDir.toString(),
+                                "iceberg.nessie-catalog.authentication.type", "BEARER",
+                                "iceberg.nessie-catalog.authentication.token", accessToken,
```

Comment:
I commented out this line to simulate forgetting to enter a bearer and got the following exception

```
2023-07-27T23:03:07.768-0600	INFO	main	io.airlift.bootstrap.LifeCycleManager	Life cycle stopped

io.trino.testing.QueryFailedException: Unauthorized (HTTP/401): 
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `org.projectnessie.error.ImmutableNessieError` from [Unavailable value] (token `JsonToken.NOT_AVAILABLE`)
 at [Source: UNKNOWN; byte offset: #UNKNOWN]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
```

while running `io.trino.plugin.iceberg.catalog.nessie.TestIcebergNessieCatalogWithAuth#testWithValidAccessToken`.


Follow a bit the logic from  

https://github.com/trinodb/trino/blob/46cad1cc30828e92990de80f5b5c23f038cfa607/lib/trino-hdfs/src/main/java/io/trino/hdfs/gcs/HiveGcsConfig.java#L72-L81 

The validation logic can probably be inlined in `io.trino.plugin.iceberg.catalog.nessie.IcebergNessieCatalogModule#createNessieIcebergClient` to fail fast when the  connector is being bootstrapped.

Side-note: specifying `iceberg.nessie-catalog.authentication.token` should be allowed only if `iceberg.nessie-catalog.authentication.type` is set to `BEARER`.